### PR TITLE
Added Save and Load Dataset methods to finish clss

### PR DESCRIPTION
**Pull request by thibmonsel**
Add 2 new functions and tests with it saveDataset() and loadDataset().

For loadDataset() : 
From the xml file we get the data and return an perftable and unordered_map from it. 

For saveDataset():
With AlternativesPerformance, PerformanceTable and the number of categories we save the content in an xml file

### CODE REVIEW

File: data/test.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <datasetName>test.xml</datasetName>
```
Comment (Mostah) [inline data/test.xml]: Can you rename this file to something related to dataset? So we know that this on is used in dataset unit test

File: src/types/DataGenerator.cpp [hunk 2]
```diff
@@ -191,44 +192,51 @@ void DataGenerator::modelGenerator(int nb_criteria, int nb_categories,
...
 std::tuple<float, Criteria, PerformanceTable>
 DataGenerator::loadModel(std::string fileName) {
+  pugi::xml_document doc = DataGenerator::openXmlFile(fileName);
+
+  if (DataGenerator::getXmlFileType(fileName) != "model") {
+    // if we have a model xml
+    throw std::invalid_argument(
+        "Cannot find any lambda and profile performances in xml file, "
+        "most likely have a dataset xml file");
+  } else {
```
Comment (Mostah) [inline src/types/DataGenerator.cpp:204]: nit: the else statement is not necessary here as if you get into the if statement the run of the function will stop anyway :)

File: src/types/DataGenerator.cpp [hunk 3]
```diff
@@ -319,6 +328,146 @@ void DataGenerator::saveModel(std::string fileName, float lambda,
   }
 }
 
+std::tuple<PerformanceTable, std::unordered_map<std::string, Category>>
+DataGenerator::loadDataset(std::string fileName) {
```
Comment (Mostah) [inline src/types/DataGenerator.cpp]: Should we return an AlternativesPerformance here instead of the tuple ?

**Review by Mostah — APPROVED** [review]
It looks good to me ! Just wondering, should we keep the commented test? Would you like to make them work in the future?

File: data/test.xml (hunk 1 above)
Comment (thibmonsel) [reply data/test.xml]: done

File: src/types/DataGenerator.cpp (hunk 2 above)
Comment (thibmonsel) [reply src/types/DataGenerator.cpp:204]: took it out

File: src/types/DataGenerator.cpp (hunk 3 above)
Comment (thibmonsel) [reply src/types/DataGenerator.cpp]: yup changed

File: test/types/TestDataGenerator.cpp [hunk 4]
```diff
@@ -7,21 +7,19 @@
...
   EXPECT_EQ(os2.str(), "1");
 }
-*/
 
 TEST(TestDataGenerator, TestDatasetGeneratorNotOverwrite) {
   DataGenerator data = DataGenerator();
   try {
-    data.datasetGenerator(3, 20, 4, "test.xml", 0, 0);
+    data.datasetGenerator(3, 5, 4, "test_dataset.xml", 0, 0);
     FAIL() << "should have throw invalid_argument error.";
```
Comment (edix9) [inline test/types/TestDataGenerator.cpp:23]: not important but throw**n**

**Pull request merged**